### PR TITLE
Display Sharing before Signup rpc as popup

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

This hooks up the incoming rpc as a popup, just like trackers.

No styling was done here, just hooking up.

screen shot 2016-05-06 at 5 02 42 pm

### CODE REVIEW

File: desktop/renderer/remote-manager.js [hunk 1]
```diff
@@ -228,8 +225,7 @@ export default connect(
     trackerStopTimer,
     trackerOnClose,
```
Comment (chrisnojima) [inline desktop/renderer/remote-manager.js]: man we should really re-discuss cleaning this file up :)

File: shared/actions/tracker.js [hunk 2]
```diff
@@ -83,6 +85,27 @@ export function registerUserChangeListener (): TrackerActionCreator {
...
+export function registerDisplayTLFCreateWithInvite (): TrackerActionCreator {
+  return dispatch => {
+    const params: incomingCallMapType = {
+      'keybase.1.identifyUi.displayTLFCreateWithInvite': (args, response) => {
+        dispatch(({payload: args, type: Constants.showNonUser}: ShowNonUser))
+        response.result()
+      }
+    }
+
+    engine.listenGeneralIncomingRpc(params)
```
Comment (chrisnojima) [inline shared/actions/tracker.js]: i was also thinking we should talk about a single place where this kind of stuff happens. like some incoming messages middleware where you can register things. right now its very hard to understand whats registering. don't have to deal with it in this PR but something to talk about

File: shared/engine/index.js [hunk 3]
```diff
@@ -283,7 +283,7 @@ class Engine {
     } else if (method === 'keybase.1.logUi.log' && this._hasNoHandler(method, callMap || {}, this._generalIncomingRpc)) {
       log(param)
       wrappedResponse.result()
-    } else if (!sessionID && this.generalListeners[method]) {
```
Comment (chrisnojima) [inline shared/engine/index.js]: why? with patrick's new commit we again have sessionID===0 meaning its incoming

File: shared/reducers/pinentry.js [hunk 4]
```diff
@@ -77,7 +77,7 @@ export default function (state: RootPinentryState = initialState, action: Pinent
       }
       return state
     default:
-      if (state.started && sessionID != null) {
+      if (state.started && sessionID != null && isPinentryAction(action)) {
```
Comment (chrisnojima) [inline shared/reducers/pinentry.js]: why is this necessary?

File: shared/engine/index.js (hunk 3 above)
Comment (MarcoPolo) [inline shared/engine/index.js]: Sorry should have mentioned this in the PR. This RPC sets a sessionID to be non-zero. I meant to mention this in #core but forgot. Will do.

File: shared/reducers/pinentry.js (hunk 4 above)
Comment (MarcoPolo) [inline shared/reducers/pinentry.js]: It's a really subtle bug.

Note that we return a modified state even if this isn't a pinentry action.

Next we don't pass in a proper initial state, just an empty object (which shouldn't type check, but does annoyingly. Can be fixed by making the type:
```
export type RootPinentryState = {
  started: boolean,
  pinentryStates: {
    [key: number]: ?PinentryState
  }
}

```
)

Then the final part of the bug is when we check to show the pinentry windows we look for the presence of `!state.closed` but since the pinentryState is just `{}` it passes that check.

This hasn't come up before because we've never passes the sessionID in the action.payload.sessionID.

Comment (MarcoPolo) [inline shared/reducers/pinentry.js]: The end result is an empty pinentry window with a broken render because it has no data.

File: desktop/renderer/remote-manager.js (hunk 1 above)
Comment (MarcoPolo) [inline desktop/renderer/remote-manager.js]: I know! what happened to the branch you were working on?

I have a vague idea to clean it up that will be ~1pt and based off the comments I left you on that branch

File: shared/constants/tracker.js [hunk 5]
```diff
@@ -36,6 +39,7 @@ export const updateProofState = 'tracker:updateProofState'
 export const reportLastTrack = 'tracker:reportLastTrack'
 
 export const onClose = 'tracker:onClose'
+export type OnClose = TypedAction<'tracker:onClose', void, void>
```
Comment (chromakode) [inline shared/constants/tracker.js]: Style question: Can you use `onClose` here to avoid repeating the constant? Worth doing?

File: shared/tracker/dumb.desktop.js [hunk 6]
```diff
@@ -209,26 +211,28 @@ const propsFiveProof: TrackerProps = {
...
-      'New user, follows me': propsNewUserFollowsYou,
-      'Followed': propsFollowing,
-      'Changed/Broken proofs user you dont follow': {...propsChangedProofs, lastTrack: false},
-      'Changed/Broken proofs': propsChangedProofs,
-      'You track them': {...propsFollowing, userInfo: {...propsNewUser.userInfo, followsYou: false}},
-      'Unfollowed': propsUnfollowed,
-      'Barely there': propsLessData,
-      'Whatevz': propsWhatevz
-    }
+const dumpMap: DumbComponentMap<Tracker> = {
```
Comment (chromakode) [inline shared/tracker/dumb.desktop.js]: dump?

**Comment by chromakode** [comment]
Nice! :+1:

File: shared/constants/tracker.js (hunk 5 above)
Comment (chrisnojima) [inline shared/constants/tracker.js]: I don't know that that works in flow?

File: desktop/renderer/remote-manager.js (hunk 1 above)
Comment (chrisnojima) [inline desktop/renderer/remote-manager.js]: i think we had some better idea of cleaning it up that was larger? i dunno. i made a ticket

**Comment by chrisnojima** [comment]
👍

File: shared/constants/tracker.js (hunk 5 above)
Comment (MarcoPolo) [inline shared/constants/tracker.js]: You can not, flow doesn't want to evaluate the js (event if its a simple const string). Maybe a future flow version will fix that but for now you have to pass in the literal string if you want it to match the literal string as the type
